game-samples/src/golang/matchmaking-service/main.go

```diff
@@ -0,0 +1,80 @@
+package main
+
+import (
+	"context"
+	"log"
+	"net/http"
+
+	spanner "cloud.google.com/go/spanner"
+	"github.com/dtest/spanner-game-match-service/models"
+	"github.com/gin-gonic/gin"
+)
+
+// Mutator to create spanner context and client, and set them in gin
+func setSpannerConnection(connectionString string) gin.HandlerFunc {
+	ctx := context.Background()
+	client, err := spanner.NewClient(ctx, connectionString)
+
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	return func(c *gin.Context) {
+		c.Set("spanner_client", *client)
+		c.Set("spanner_context", ctx)
+		c.Next()
+	}
+}
+
+// Helper function to retrieve spanner client and context
+func getSpannerConnection(c *gin.Context) (spanner.Client, context.Context) {
+	return c.MustGet("spanner_client").(spanner.Client),
+		c.MustGet("spanner_context").(context.Context)
+}
+
+// Creating a game assigns a list of players not currently playing a game
+func createGame(c *gin.Context) {
+	var game models.Game
+
+	client, ctx := getSpannerConnection(c)
+	gameID, err := models.CreateGame(game, ctx, client)
```

Comment:
nit: idiomatic Go style says that ctx should be the first argument to the function. Also, I don't really understand why we are passing in a `Game` struct here. It is passed by value, so any changes that are made by the `CreateGame` function will not be visible to the caller.

Reply:
great feedback. There's no reason for `game` to be passed in with current code. I may change this to be passed by reference and not return the gameID. That way the endpoint can be modified in the future to provide other info about the game back to the caller.